### DISCUSSION TITLE
Remote Desktop Hanging when launching new or old Stellarium

### DISCUSSION THREAD

**Discussion by TheTurntableSlave** [Stellarium/stellarium#2749] [Q&A] [answered]
Hi all

I've been having a few issues with Stellarium on my remote computer recently. I've upgraded both my laptop and desktop to W11 22H2, and whenever I launch Stellarium (both "normal" and angle mode) on one of my remote computers, the remote session hangs. I have to close it down, then re-open the session and when I do, Stellarium has (presumably) crashed.

Note I have tested on both the new 1.0 release and the older 0.22x release with the same issue.

When this happened a while ago, I would switch to angle mode as I know there are some RDP issues running graphics intensive programs and Angle mode fixed that. Now even that doesn't work.

The program is installed on a W10 box. I've attached the log.
Note I also have another Astro PC running Windows 11 and connecting to that PC and running Stellarium works fine. It's only when I connect either of my laptop/PC to the W10 box where things go wrong.

I've also got a W11 server and I tried to connect that to the W10 box and launch Stellarium, and I get the same issue.

PC specs of the W10 box in question:
Bee-Link U55, W10 Pro (19044), 8GB RAM, i3-5005U, 256GB SSD.

And just if you need it:
PC; Ryzen 3900X, 32GB RAM, RTX 2070S, W11 Pro (22621)
Laptop: Surface Book 2, i7 8650U, 16GB RAM, GTX 1050, W11 Pro (22621)
Server: i5-9400, 16GB RAM, W11 Pro (22000)

AstroPC 1: Bee-link above. This is the one that gives me issues (but only within the past few days, worked fine previously)
AstroPC 2: Miele Quieter 3Q, W11 Pro (22621), Celeron N5105, 8GB RAM, 

the attached logs are for launching both Angle and non-Angle modes of Stellarium

Any help appreciated! Thanks :)

[log2.txt](https://github.com/Stellarium/stellarium/files/9793403/log2.txt)
[log.txt](https://github.com/Stellarium/stellarium/files/9793404/log.txt)

**Comment by alex-w** [comment, marked as answer]
The bad news - both logs show that your PC is using Intel(R) HD Graphics 5500 and ANGLE mode is not working for Qt6 environment anymore. So, you need use Qt5-based Stellarium to enable ANGLE mode (e.g., version 0.22.3) or install discrete graphics card (with drivers of course) to use full OpenGL features (at least Qt think that Intel GPU is had bad drivers).

**Reply by gzotti** [reply]
Also please note that the (ANGLE) links don't work any longer. You must set environment variables as described in User Guide, section 6.2.2.

**Reply by TheTurntableSlave** [reply]
that's just it. The same thing happens in 0.22.3
That's why I upgraded to the new 1x release, as I thought I could fix it with a new install.
Sadly a discrete GPU is out of the question for this mini-PC that sits on my telescope.

I've just uninstalled the 1x and reinstalled 0.22.2, and that works. Can I also ask @gzotti , do you mean ANGLE links don't work for ALL versions? Because indeed it seems I get the same crash/hang on 0.22.2 when opening ANGLE. When I load the standard version it seems to work.

**Reply by gzotti** [reply]
The link in the start menu which says "Stellarium (ANGLE mode)" does not switch to ANGLE mode. The solution is my above sentence. This is also valid for Qt5-based V1.0 (0.22.3). Your U55 system should not be used with Qt6-based V1.0 (1.23.3). The three other PCs should be good for 1.22.3.

**Reply by gzotti** [reply]
I must add I don't use Stellarium over any RDP, so I have no experience, and we know nobody who would ever have written a review of probably ~25 PC systems of different age and price tag w.r.t. Stellarium, Telescope control and RDP.

**Reply by TheTurntableSlave** [reply]
thanks for the info. I'll mark this as resolved and just stick to 0.22.2 on my U55. Maybe one day I'll get a U59 which uses the same processor as the MeLE.
btw, lots of people use Stellarium over RDP for astrophotography at least.

**Reply by 10110111** [reply]
@TheTurntableSlave Could you elaborate on the use case of RDP? Do you run an RDP client on a smartphone to connect to a PC at home? I wonder because I'd normally just take a laptop with Stellarium running on it directly.

**Reply by gzotti** [reply]
A contemporary development of even amateur observing is having Stellarium on a little PC in the cold observatory dome (or exterior pillar) and the operator sitting in the warm kitchen 100m or even 4000km away. I doubt that a real-time graphics heavy program is the best solution for such scenario, but it is being increasingly reported.

**Reply by TheTurntableSlave** [reply]
> @TheTurntableSlave Could you elaborate on the use case of RDP? Do you run an RDP client on a smartphone to connect to a PC at home? I wonder because I'd normally just take a laptop with Stellarium running on it directly.

Sure @10110111 . The standard approach if using a PC for controlling a telescope is to install Stellarium on said PC and attach it on or near the telescope. The reason this is better than a laptop (I used to use a laptop previously) is
1. you don't need to be near the telescope for imaging - you can remote into the mini-PC and setup everything from inside your home/test/car/whatever. You'd just need to be next to it if you're polar aligning each time.
2. if your PC is connected to the scope, you can cable manage a lot better, rather than having cables dropping to the ground or wherever your laptop is located. Cable management is very important! 

For RDP, I use any device to connect to the mini-PC... I've used an iPad, my phone, my laptop, and my PC. Whatever is available. 

Example of my current setup, attached. Mini PC up top along with a Pegasus unit for power management.  I can image from inside the house without having to be near the scope. I use Stellarium to frame my target, and then send that framing/rotation information to NINA which drives the imaging session.
Of course, NINA has its own rotation/framing ability, but I prefer Stellarium with its deeper library and easier GUI.

![20221002_074330683_iOS](https://user-images.githubusercontent.com/94218986/196065542-92d081bd-4b7e-403f-ba32-9ee8962275eb.jpg)
